tests/e2e/data_masking/test_data_masking.py

```diff
@@ -0,0 +1,129 @@
+    return infrastructure.get("BasicHandler", "")
+
+
+@pytest.fixture
+def basic_handler_fn_arn(infrastructure: dict) -> str:
+    return infrastructure.get("BasicHandlerArn", "")
+
+
+@pytest.fixture
+def kms_key1_arn(infrastructure: dict) -> str:
+    return infrastructure.get("KMSKey1Arn", "")
+
+
+@pytest.fixture
+def kms_key2_arn(infrastructure: dict) -> str:
+    return infrastructure.get("KMSKey2Arn", "")
+
+
+@pytest.fixture
+def data_masker(kms_key1_arn) -> DataMasking:
+    return DataMasking(provider=AwsEncryptionSdkProvider(keys=[kms_key1_arn]))
+
+
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption(data_masker):
+    # GIVEN an instantiation of DataMasking with the AWS encryption provider
+
+    # AWS Encryption SDK encrypt method only takes in bytes or strings
+    value = bytes(str([1, 2, "string", 4.5]), "utf-8")
+
+    # WHEN encrypting and then decrypting the encrypted data
+    encrypted_data = data_masker.encrypt(value)
+    decrypted_data = data_masker.decrypt(encrypted_data)
+
+    # THEN the result is the original input data
+    assert decrypted_data == value
+
+
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption_context(data_masker):
+    # GIVEN an instantiation of DataMasking with the AWS encryption provider
+
+    value = bytes(str([1, 2, "string", 4.5]), "utf-8")
+
+    # WHEN encrypting and then decrypting the encrypted data with an encryption_context
+    encrypted_data = data_masker.encrypt(value, encryption_context={"this": "is_secure"})
+    decrypted_data = data_masker.decrypt(encrypted_data, encryption_context={"this": "is_secure"})
+
+    # THEN the result is the original input data
+    assert decrypted_data == value
+
+
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption_diff_context_fail(data_masker):
+    # GIVEN an instantiation of DataMasking with the AWS encryption provider
+
+    value = bytes(str([1, 2, "string", 4.5]), "utf-8")
+
+    # WHEN encrypting with a encryption_context
+    encrypted_data = data_masker.encrypt(value, encryption_context={"this": "is_secure"})
+
+    # THEN decrypting with a different encryption_context should raise a ValueError
+    with pytest.raises(ValueError):
+        data_masker.decrypt(encrypted_data, encryption_context={"not": "same_context"})
+
+
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption_no_context_fail(data_masker):
+    # GIVEN an instantiation of DataMasking with the AWS encryption provider
+
+    value = bytes(str([1, 2, "string", 4.5]), "utf-8")
+
+    # WHEN encrypting with no encryption_context
+    encrypted_data = data_masker.encrypt(value)
+
+    # THEN decrypting with an encryption_context should raise a ValueError
+    with pytest.raises(ValueError):
+        data_masker.decrypt(encrypted_data, encryption_context={"this": "is_secure"})
+
+
+# TODO: metaclass?
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption_key_fail(data_masker, kms_key2_arn):
```

Comment:
```suggestion
def test_encryption_decryption_key_mismatch(data_masker, kms_key2_arn):
```

nt: slightly clearer intent